web3/utils/abi.py

```diff
@@ -1,21 +1,115 @@
 from typing import (
-    List,
+    Any,
+    Callable,
+    Optional,
+    Sequence,
+    Tuple,
     Union,
 )
 
-from web3.types import (
-    ABIEvent,
+from eth_abi.abi import (
+    ABICodec,
+)
+from eth_typing import (
+    ABIConstructor,
+    ABIFallback,
     ABIFunction,
+    ABIReceive,
+    HexStr,
+    TypeStr,
+)
+from eth_utils.abi import (
+    get_abi_input_types,
+)
+from eth_utils.hexadecimal import (
+    encode_hex,
+)
+from hexbytes import (
+    HexBytes,
+)
+
+from web3._utils.abi import (
+    build_non_strict_registry,
+    build_strict_registry,
+    check_if_arguments_can_be_encoded,
+    map_abi_data,
 )
+from web3._utils.encoding import (
+    to_hex,
+)
+from web3._utils.normalizers import (
+    abi_address_to_hex,
+    abi_bytes_to_bytes,
+    abi_string_to_text,
+)
+
+
+def encode_abi(
+    function_abi: Union[ABIFunction, ABIConstructor, ABIFallback, ABIReceive],
+    arguments: Sequence[Any] = None,
+    data: Optional[HexStr] = None,
+    is_async: Optional[bool] = False,
+    data_normalizers: Optional[Sequence[Callable[..., Tuple[TypeStr, Any]]]] = None,
+    abi_codec: Optional[ABICodec] = None,
+    strict: Optional[bool] = True,
+) -> HexStr:
+    """
+    Return encoded data from a function ABI and arguments.
+    :param function_abi: Function ABI.
+    :param type: `ABIFunction`
```

Comment:
If the above comment is correct, this should change as well.

```suggestion
    :param type: `ABICallable`
```